### DISCUSSION TITLE
BMW/MINI Connected 6.5 no longer connects to the car

### DISCUSSION THREAD

**Discussion by hufman** [BimmerGestalt/AAIdrive#238] [Q&A]
It seems that BMW (and Mini) Connected 6.5 no longer starts the BMW Apps connection, and instead shows a popup to switch to the new My BMW app. The previous 6.4 version still works and is available [here (European)](https://apkpure.com/bmw-connected/de.bmw.connected/versions) or [here (US)](https://apkpure.com/bmw-connected/de.bmw.connected.na/versions), and seems to still be reliable. Remember to disable automatic updates in the Play Store if you downgrade the app.

It appears that recent versions of My BMW do provide the apps connection. Any [nightly AAI build](https://androidautoidrive.s3.amazonaws.com/hufman/AndroidAutoIdrive/androidautoidrive-latest-master-nomap-sentry-release.apk) after November is capable of using this directly, and the recent 1.3 Release Candidates are also well supported, but earlier versions (including the release from August) also need BMW/Mini Connected (or Classic) installed to provide the security service.

MyBMW does not provide integration to launch 3rd party apps, which AndroidAutoIdrive relied on to launch automatically, so AAI now also attempts to launch when the Bluetooth audio connection to the car starts.

Connected 6.4 is still functional for now, and its Apps connection still works even if the app itself isn't logged in, though Calendar access might be harder to grant. Reimplementing a calendar for MyBMW is scheduled in #279. USB support in MyBMW seems less reliable than Connected, and hopefully this continues to improve over time. Please use the feedback feature in MyBMW to request BMW to prioritize the features you want to see.

**Comment by fabssor** [comment]
That's Strange. I am running BMW connected App 6.5 .
Playstore even says it was last updated in November.
My setup works just fine with that version. Spotify and all the other apps show up as expected :)

**Reply by wdata68** [reply]
It's probably related to the fact that BMW started new MyBMW application.  I noticed that there is no connection anymore yesterday, tried everything to fix it, and ended up with cleaning all the data of BMW Connected in the morning. After the initial configuration there were a warning saying the BMW Connected is to be deactivated soon because of new MyBMW application. 
I'm going to try it today later after leaving the office.

**Reply by fabssor** [reply]
So I was wrong. My setup did not work. Even though the playstore Said that the connected App was last updated in November.

But good News! The new MyBMW App works now🎉😀 Connection works ans androidAutoIdrive establishes the Connection!

**Reply by unlive74** [reply]
Bad news. androidAutoIdrive cannot establish a connection with the car (X2) automatically when I use the new MyBMW App. Rollback to 6.4

**Reply by hufman** [reply]
How curious! My understanding is that 6.5 came out in January, but perhaps that's just when it was uploaded to apkpure?
It would be great if MyBMW supported the protocol still!

**Comment by kalahcustom** [comment]
I had the same issue yesterday. Connected app was last updated in December. However I was not able to connect. Bluetooth settings while phone was connected didn't show the "applications" connection. It was available only for telephone and music. Apps square simply disappeared.

However I wiped out all the Bluetooth settings in my car as well as in my phone, deleted all bmw apps and did a restart of Bluetooth in the car as well. I installed only new MyBMW app and established all connections from the beginning. Android Auto iDrive showed connection again and everything works with new app as well.

I faced only one issue so far. Phone doesn't want to connect automatically with apps, so I have to reconnect it manually from the car each time. My method with constantly locked apps doesn't work in this case.

**Reply by unlive74** [reply]
> I faced only one issue so far. Phone doesn't want to connect automatically with apps, so I have to reconnect it manually from the car each time. My method with constantly locked apps doesn't work in this case.

To connect to the car using My BMW now, I just need to launch androidAutoIdrive (rev. 23/03/21) on my phone. But I have to do this every time I start the car.

**Reply by fabssor** [reply]
Tested it also again and after starting androidAutoIdrive manually everything (have to do it evety time) works smooth.

**Reply by hufman** [reply]
Thanks for the information! I'll see what I can do about improving automatic startup!

**Reply by kalahcustom** [reply]
I'll give a try to automate opening AAI with bixby routine while connecting by Bluetooth to the car. In the meantime I'll keep myBMW in lock mode. I hope this temporary solution will work

**Reply by hufman** [reply]
Good news! In some brief experiments over lunch, I found that the equivalent My Mini app (tested version 1.3.2) now does connect to my car and lets AAI run! So I have a path forward to test out solutions for this.

After Connected Classic's series of internal apps, Connected added the Car API feature to allow external apps like Spotify to be added to the car. It does this by sending an announcement out to the phone to ask these compatible apps to announce themselves to be saved in a list inside Connected. When the car connects, Connected goes through this list and starts up each one directly.
AndroidAutoIdrive uses this same process to register itself to be started by Connected. From what I see in the MyBMW app, the information stored about each app now includes a "status" flag for whether the app is ACTIVATED or other statuses, such as if the app needs to be signed in first, and it only starts ACTIVATED apps when connecting to the car.

I tried a quick test to add this ACTIVATED flag in AndroidAutoIdrive and it wasn't enough, but this is the fun part :) I'll take some more time to enable the debug logs in My BMW and see exactly what's going on to get a tighter integration.

---

However, I wanted Connected Classic users to experience the same automatic startup that this CarAPI provides for Connected users. When Connected and Connected Classic start up, they send a broadcast out to the phone saying a car is connected, and while I can't fully start the app from stopped with that signal, I do [listen for this signal](https://github.com/hufman/AndroidAutoIdrive/blob/445fb777c866b15227c0f5b749406c86cde511e2/app/src/main/java/me/hufman/androidautoidrive/notifications/NotificationListenerServiceImpl.kt#L63-L68) and start up the connection logic from within the Notification Listener if it is enabled and running.

When I did my tests, I found that while I had the Notification Listener enabled, the My Mini app did successfully start the AAI app, and I needed to remove Notification Access to reproduce the same behavior that you encountered. I believe (haven't checked deeper in the code to confirm) that My BMW/Mini is using the [same old original signal](https://github.com/hufman/IDriveConnectKitAndroid/blob/e0de3d6b47f42403d9819c54b8c5f2c77d38c738/src/main/java/me/hufman/idriveconnectionkit/android/IDriveConnectionListener.kt#L48) as Connected Classic to start up its own internal processing, and so it wakes up AAI too while this Notification Listener is running. Can you confirm that you are not running with Notifications Access? If you are running Notifications Access and MyBMW was still unable to start AAI, then my theory is less valid, but otherwise this might provide a workaround until I figure out the correct solution!

**Reply by fabssor** [reply]
> Good news! In some brief experiments over lunch, I found that the equivalent My Mini app (tested version 1.3.2) now does connect to my car and lets AAI run! So I have a path forward to test out solutions for this.
> 
> After Connected Classic's series of internal apps, Connected added the Car API feature to allow external apps like Spotify to be added to the car. It does this by sending an announcement out to the phone to ask these compatible apps to announce themselves to be saved in a list inside Connected. When the car connects, Connected goes through this list and starts up each one directly.
> AndroidAutoIdrive uses this same process to register itself to be started by Connected. From what I see in the MyBMW app, the information stored about each app now includes a "status" flag for whether the app is ACTIVATED or other statuses, such as if the app needs to be signed in first, and it only starts ACTIVATED apps when connecting to the car.
> 
> I tried a quick test to add this ACTIVATED flag in AndroidAutoIdrive and it wasn't enough, but this is the fun part :) I'll take some more time to enable the debug logs in My BMW and see exactly what's going on to get a tighter integration.
> 
> However, I wanted Connected Classic users to experience the same automatic startup that this CarAPI provides for Connected users. When Connected and Connected Classic start up, they send a broadcast out to the phone saying a car is connected, and while I can't fully start the app from stopped with that signal, I do [listen for this signal](https://github.com/hufman/AndroidAutoIdrive/blob/445fb777c866b15227c0f5b749406c86cde511e2/app/src/main/java/me/hufman/androidautoidrive/notifications/NotificationListenerServiceImpl.kt#L63-L68) and start up the connection logic from within the Notification Listener if it is enabled and running.
> 
> When I did my tests, I found that while I had the Notification Listener enabled, the My Mini app did successfully start the AAI app, and I needed to remove Notification Access to reproduce the same behavior that you encountered. I believe (haven't checked deeper in the code to confirm) that My BMW/Mini is using the [same old original signal](https://github.com/hufman/IDriveConnectKitAndroid/blob/e0de3d6b47f42403d9819c54b8c5f2c77d38c738/src/main/java/me/hufman/idriveconnectionkit/android/IDriveConnectionListener.kt#L48) as Connected Classic to start up its own internal processing, and so it wakes up AAI too while this Notification Listener is running. Can you confirm that you are not running with Notifications Access? If you are running Notifications Access and MyBMW was still unable to start AAI, then my theory is less valid, but otherwise this might provide a workaround until I figure out the correct solution!

If you mean the notification access in the Android settings (search for half an hour to find that stupid setting again 😅), then it is (for me) activated for AAiD. 
MyBMW and AAiD also have full permissions.
If you need any more testing, don't hesitate to ask! Cheers

**Reply by sawinco** [reply]
Can I just qualify something, do I only need the new my mini app to get auto running. When I run it, in the connection section, it recognises the new app is installed but says its not supported. Using latest overnight build 1.2.2-403-445fb77
Thanks

**Reply by fabssor** [reply]
So again some notes from me. 
Today it did not work... Could Not get AAiD to Connect to MyBMW. The tunnel always failed with Handshake failed.

**Reply by sawinco** [reply]
Thanks fabssor, tried it a few times today with the equivalent new my mini app and got AAid to connect. I use tasker to start all the apps on Detection of connection to the cars Bluetooth.

**Reply by kalahcustom** [reply]
So I tried with bixby routine today. AAiD connects for a second than disconnects. It shows Spotify for a second than it disappears. So for now I have to launch manualy AAiD every time I start the car to make it work.

**Reply by hufman** [reply]
Thank you for the offer to help test! As soon as I have a test branch build, I'll post it here.

That notice about "My BMW is not supported" is apparently outdated, I'll clarify that as part of this integration fix. The new advice is to not have both My BMW and BMW Connected installed at the same time.

The Handshake Failed message is produced directly by the BMW app (that BCL Tunnel panel is just relaying the debug information from BMW), perhaps try toggling the car's Music or Apps settings to get past this.

A Bixby routine does seem to work for some people, but I can't give much more advice than that. The fact that Spotify showed up in the car is a good sign. I have experienced a few instances with Connected where it starts up, tears down, and starts up again the car connection, and if My BMW did this (without being able to launch AAI) it would look like what you saw. Hopefully it's relatively rare, and the Bixby automation works most of the time!

**Reply by hufman** [reply]
I've uploaded a [build to test a workaround](https://androidautoidrive.s3.amazonaws.com/hufman/AndroidAutoIdrive/androidautoidrive-latest-bt_launch-nomap-sentry-release.apk). This method attempts to start the app's MainService when the Bluetooth connection to the car connects, and then it will do the same probing process as when opening the UI to manually start the connection.
I'd greatly appreciate your assistance in testing this approach, especially if you are using only MyBMW and the automatic connection isn't working for you, especially if you didn't enable Notification Access (which includes a separate workaround already). The method I'm using to start the MainService from the background needs me to show a notification, and I technically do, but then hide it right away until the connection is finalized, and so I'd like confirmation that this approach works well.

So far, it feels like My BMW connects the Bluetooth Apps to the car faster than BMW Connected, has anyone else noticed this too?

**Reply by unlive74** [reply]
Thanks! It works great! It is necessary to correct in UI "Unsupported application My BMW is installed".

**Reply by sawinco** [reply]
In terms of the Bluetooth apps connecting faster I can confirm that seems to be the case in the my mini app. Not sure but connection seems more stable as well. I'll try the new build and turn off tasker starting aai to see if it will start

**Reply by unlive74** [reply]
By the way, it seems to me that now the connection with my car (X2, EntryEvo ID5, your app recognizes it as BMW ID6L) is more often without applications (only the phone and bluetooth audio). Previous version + Connected 6.4 = 90% success, Current version + My BMW = 50/50. When reconnecting, it's usually corrected.

**Reply by hufman** [reply]
I updated this branch's build to improve the UI text about MyBMW being supported, check it out! I think it's pretty much ready to merge, if it all checks out.

**Reply by unlive74** [reply]
> I updated this branch's build to improve the UI text about MyBMW being supported, check it out! I think it's pretty much ready to merge, if it all checks out.

I had to go back to Connected 6.4 - it's impossible when you have to reconnect to the car at 50%

**Reply by sawinco** [reply]
Limited testing with my mini app. Aai seems to auto start when connected app has established connection with car, the apps connection seems quicker and more stable. Had a few instances where the AAI tunnel took a few instances to establish but need to try it a bit more too see if I can seen any pattern to this. So far so good 😁

**Comment by HaivanJV** [comment]
I can establish the connection with MyBMW, however I experienced several app crashes and "App" disconnections .
From my side, I believe until a few connectivity issues are addressed and corrected I will use BMW Connected 6.4
Did you experience the same flaky connectivity with MyBMW?

**Reply by hufman** [reply]
When I last looked into MyBMW a few months ago, they hadn't enabled the Apps protocol through it, so all these reports about it working now are a big surprise to me!

**Reply by drm87** [reply]
> I can establish the connection with MyBMW, however I experienced several app crashes and "App" disconnections .
> From my side, I believe until a few connectivity issues are addressed and corrected I will use BMW Connected 6.4
> Did you experience the same flaky connectivity with MyBMW?

Do you mean it works when you have only the "MyBMW" app installed, or when having both "MyBMW" and "BMW Connected" apps installed?

**Reply by unlive74** [reply]
> Do you mean it works when you have only the "MyBMW" app installed, or when having both "MyBMW" and "BMW Connected" apps installed?

It works when I have only the "MyBMW" app installed, but it doesn't work when I have both "MyBMW" and "BMW Connected" apps installed

**Reply by fabssor** [reply]
A while ago I asked @hufman about the MyBMW App and if it will be integrated (see #149). At this time it did not work. But as I remember correctly both Apps fight over the connection and AAiD did crash because of that. 
So be sure you only have installed the MyBMW App.

**Reply by HaivanJV** [reply]
@masteralarm I had only the MyBMW installed. What I experienced was mainly a drop of the "App" tunnel and crashes.
Also, I experienced the same as other user did, with having to open AAID manually to get the list of the apps out on the iDrive

**Comment by desertspotter** [comment]
First, I would like to say, what an awesome job hufman!!!!

I have the same problem a few days ago, I was using spotify version 8.5.66.1002 (last with bmw connected support) with connected app but it stops connecting with the car and with the new  "my bmw" app it doesn't work. I tried the new alexa funtionality that the lastest my bmw app bring with my M21 F22 and it can start amazon music but no art neither playlist in idrive. 

Searching google I ended up here and what a discovery!, I was able to use your app with poweramp and spotify with connected 6.4 and it works like a charm. I downloaded the 1.2.2 release so no album art in split screen but I see other posts with the required build and I just installed it and I will test tomorrow. 

I'm looking forward future releases with "my bmw" full support (I guess the new alexa functionally  for android enables the app tunnel in it).

**Comment by mbonus** [comment]
BMW must have flipped something on the server side because 6.5 was working great and then the notification came up to "upgrade" to myBMW and the I could no longer get apps to push.  BMW id4++ headunit.  This is one that only pushes apps over usb (no bluetooth support).

Using build version 1.22.399 tried myBMW - connected via bluetooth but not usb.  Tried 6.5 BMW connected no luck.  Rolled back to 6.4(US) and all is good again.  6.5 confirmed not working again when play store auto-updated 6.4 :D 

So for now - myBMW app no good for id4++ over usb

**Reply by hufman** [reply]
Over the weekend, I discovered that my car was able to connect over usb when running version 1.4.3 of the official car app. Can you confirm if the current MyBMW app works with your ID4 headunit again?

**Comment by mbonus** [comment]
I'm still testing, it works but sporadically and I haven't isolated why or
when it works.  I'll reply again if I see a pattern.  Thanks, Mike

**Comment by mbonus** [comment]
Mybmw works consistently with a specific order:. Phone bt off, connect USB,
let aa load, turn on bt, connect phone to car via bt.

**Comment by Cross199104** [comment]
I've found out, that Connected Version 3.1.0.3071 works perfect in my F11 2014 NBT. I have it on a Galaxy s10e.
Connected starts but only shows that the server is not reachable.
But thats no problem because AAIdrive works perfectly.

Thanks Walter for your effort the excellent app and your support.

**Comment by ramawaad-sys** [comment]
Hi BimmerGestalt Team,

First of all, thank you for the incredible work on AAIdrive — it is truly the best solution available for BMW owners who want Android features without expensive hardware modifications.

I am writing to request a few feature improvements that would make AAIdrive even more powerful:

1. 🗺️ Google Maps & Waze Integration
Currently, the map view in AAIdrive is limited and slow over Bluetooth. It would be amazing if Google Maps and Waze could be displayed directly and smoothly on the iDrive screen — similar to how Android Auto handles navigation. Many BMW F-Series owners with iDrive 5/6 (especially F45, F30, F10) are waiting for this feature.

2. 🚨 Blitzer.de / Speed Camera Alerts
Blitzer.de is the most popular speed camera and radar warning app in Germany and Europe. Integrating Blitzer.de alerts into the iDrive display — even as simple on-screen notifications or instrument cluster warnings — would be extremely valuable for European users.

3. 🔄 MyBMW 6.x Compatibility Fix
Recent versions of MyBMW (6.x) no longer connect to the car for AAIdrive. Many users are stuck downgrading to older versions. An official workaround or updated connection method compatible with MyBMW 6.5+ would greatly improve the user experience.

I drive a BMW 225xe iPerformance (F45, 2017, iDrive 6 / EntryNav2) with wireless CarPlay activated. Like many Android users with BMW F-Series cars, AAIdrive is our only real option for smart phone integration.

Thank you for considering these requests. AAIdrive is a fantastic project and deserves wider recognition. I would also be happy to contribute to testing on the F45 platform.

Best regards,
A loyal AAIdrive user from Germany 🇩🇪
